forum/users/models.py

```diff
@@ -1,3 +1,52 @@
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
 from django.db import models
+from django.utils import timezone
 
-# Create your models here.
+
+class CustomUserManager(BaseUserManager):
+    def create_user(self, email, password, **extra_fields):
+        if not email:
+            raise ValueError("The Email must be set")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        extra_fields.setdefault("is_active", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError("Superuser must have is_staff=True.")
+        if extra_fields.get("is_superuser") is not True:
+            raise ValueError("Superuser must have is_superuser=True.")
+        return self.create_user(email, password, **extra_fields)
+
+
+class CustomUser(AbstractBaseUser):
+    class Role(models.IntegerChoices):
+        INVESTOR = 1
```

Comment:
I think there are no needed such roles, cause `Investors` and `Startup` will have a foreign key to `Users`.

But we can have roles such as (for the future purposes, for example):
`user` - it means that this user might have investors or startups `profiles`
`admin` - has more low-level access to the data, and has the ability to add some `Industries` (for instance)
`support` - something else

Reply:
Done.